### PR TITLE
Add selectbox for server name

### PR DESCRIPTION
**Pull request by ossinkine**
I've added select box for selecting server name but not just a text field.

I've implemented it for MongoDB only because PDO stores SERVER data in a text field and I have no idea how to select one of parameters from it.

Also an index is required for performance purpose.

### CODE REVIEW

**Comment by ossinkine** [comment]
@glensc if this is ok for you I'll try to add tests

File: templates/runs/list.twig [hunk 1]
```diff
@@ -30,7 +30,16 @@
         <div class="control-group span4">
             <label class="control-label" for="server-name">Server Name</label>
             <div class="controls">
-                <input type="text" id="server-name" name="server_name" value="{{ search.server_name }}">
+                {% if server_names is iterable %}
+                    <select id="server-name" name="server_name">
+                        {% for server_name in server_names %}
+                            <option></option>
+                            <option value="{{ server_name }}"{% if server_name == search.server_name %} selected{% endif %}>{{ server_name }}</option>
```
Comment (glensc) [inline templates/runs/list.twig:37]: is htmlspecialchars needed here?

Comment (ossinkine) [reply templates/runs/list.twig:37]: Twig escapes such symbols automatically

**Comment by glensc** [comment]
Are you able to rebase against current 0.20.x, if so, add row to compatibility matrix that was added to readme in https://github.com/perftools/xhgui/pull/461

if not, I'll rebase myself and force push its branch

**Comment by ossinkine** [comment]
I'll do

**Comment by ossinkine** [comment]
I've added a test for a new searcher functionality and also added a row to the matrix you requested

File: README.md [hunk 2]
```diff
@@ -271,11 +272,13 @@ storage.)
 | Searcher::getAllWatches()       | ✓        | ✓ [#435] |
 | Searcher::truncateWatches()     | ✓        | ✓ [#435] |
 | Searcher::stats()               | ✗ [#305] | ✓        |
+| Searcher::getAllServerNames()   | ✓        | ✗ [#460] |
```
Comment (glensc) [inline README.md]:
```suggestion
| Searcher::getAllServerNames()   | ✓ [#460]       | ✗  |
```

File: src/Searcher/SearcherInterface.php [hunk 3]
```diff
@@ -140,4 +140,11 @@ public function truncateWatches();
      * @return array array of stats
      */
     public function stats();
+
+    /**
+     * Get all the known server names.
+     *
+     * @return array|false array of server names or false if not supported
```
Comment (glensc) [inline src/Searcher/SearcherInterface.php]: i think `array|null` is better typing, then you can use `?array` as native return type

Comment (ossinkine) [reply src/Searcher/SearcherInterface.php]: I didn't add native type because it's not used in other methods

**Comment by ossinkine** [comment]
@glensc Please review

File: src/Searcher/SearcherInterface.php (hunk 3 above)
Comment (glensc) [reply src/Searcher/SearcherInterface.php]: please add the type as asked.

current minimum PHP version is 7.2 so all language features compatible with that version can be used.

Comment (ossinkine) [reply src/Searcher/SearcherInterface.php]: Updated

**Comment by ossinkine** [comment]
@glensc when do you plan to release 0.20.1 to be able to use this feature?

**Comment by glensc** [comment]
I was hoping someone helps out with automatic releases:
- https://github.com/perftools/xhgui/issues/427

**Pull request merged**